### DISCUSSION TITLE
Is Go driver thread safe to be used from few goroutines?

### DISCUSSION THREAD

**Discussion by ribtoks** [ClickHouse/clickhouse-go#1361] [Q&A] [answered]
Are there any best practices or limitations I should know about when using `clickhouse-go` driver (as "stdlib" version) from multiple goroutines to batch-insert data?

**Comment by jkaflik** [comment, marked as answer]
@ribtoks 

`clickhouse-go` is not thread-safe guaranteed, however, there shouldn't be any consequence from using it across multiple goroutines. 
This is true as long you don't reuse single `Batch` (`clickhouse-go` API) or single `Tx` (`database/sql`) across multiple goroutines.

**Reply by ribtoks** [reply]
Thank you!
